src/warc2zim/converter.py

```diff
@@ -388,15 +407,46 @@ def run(self):
 
         self.creator.finish()
 
+    def extract_warcinfo(self, record):
+        """Extract the software value from a warcinfo record"""
+        if self.warc_software:
+            logger.debug("warc_software already set, ignoring warcinfo record")
+            return
+        if get_record_mime_type(record) != "application/warc-fields":
+            logger.warning(
+                f"Unsupported warcinfo record found: {get_record_mime_type(record)}"
+            )
+            return
+        for warcfield in get_record_content(record).decode("UTF-8").splitlines():
+            name_and_value = warcfield.split(":")
+            if len(name_and_value) != 2:  # noqa: PLR2004
+                logger.warning(f"Unrecognized WARC field: {warcfield}")
+                continue
+            if name_and_value[0].strip().lower() != "software":
+                continue
+            self.warc_software = "," + name_and_value[1].strip()
```

Comment:
this is ugly. you should handle the concatenation properly when setting `Scraper`

Reply:
I have to admit I'm mostly ashamed, especially when I see the code setting `Scraper` metadata which is mostly as ugly.